### ISSUE TITLE
Offset error in article field on migration

### ISSUE THREAD

**Issue by espensgr** [barrelstrength/craft-sprout-forms#565]
Migrating the Article fieldtype in a form from Craft 2 to Craft 3 results in an offset error in the `{{craft.sproutForms.displayForm(handle)}}`.

### Steps to reproduce

Update Craft from 2 to 3

### Additional info

- Craft version: 3.5.19.1
- PHP version: 7.4.2
- Database driver & version:  MySQL 5.7.26
- Plugins & versions: Sprout Forms 3.13.2.3

Error:
```
PHP Notice 'yii\base\ErrorException' with message 'Undefined offset: 0'

in /projectpath/vendor/barrelstrength/sprout-forms/src/services/FrontEndFields.php:196

Stack trace:
#0 /projectpath/vendor/craftcms/cms/src/web/ErrorHandler.php(76): yii\base\ErrorHandler->handleError(8, 'Undefined offse...', '/Users/espensol...', 196)
#1 /projectpath/vendor/barrelstrength/sprout-forms/src/services/FrontEndFields.php(196): craft\web\ErrorHandler->handleError(8, 'Undefined offse...', '/Users/espensol...', 196, Array)
#2 /projectpath/vendor/barrelstrength/sprout-forms/src/services/FrontEndFields.php(51): barrelstrength\sproutforms\services\FrontEndFields->getSinglesEntries()
#3 /projectpath/vendor/barrelstrength/sprout-forms/src/fields/formfields/Entries.php(81): barrelstrength\sproutforms\services\FrontEndFields->getFrontEndEntries(Array)
#4 /projectpath/vendor/barrelstrength/sprout-forms/src/web/twig/variables/SproutFormsVariable.php(222): barrelstrength\sproutforms\fields\formfields\Entries->getFrontEndInputHtml(Object(craft\elements\db\EntryQuery), Object(barrelstrength\sproutforms\elements\Entry), Array)
#5 /projectpath/vendor/twig/twig/src/Extension/CoreExtension.php(1499): barrelstrength\sproutforms\web\twig\variables\SproutFormsVariable->displayField(Object(barrelstrength\sproutforms\elements\Form), Object(barrelstrength\sproutforms\fields\formfields\Entries), Array)
#6 /projectpath/vendor/craftcms/cms/src/helpers/Template.php(106): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(barrelstrength\sproutforms\web\twig\variables\SproutFormsVariable), 'displayField', Array, 'method', false, false)
#7 /projectpath/storage/runtime/compiled_templates/87/87a7f16078170ae86f5f271d9c44a99c55386f93a3bf0b12e3cff5cd894113c1.php(63): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(barrelstrength\sproutforms\web\twig\variables\SproutFormsVariable), 'displayField', Array, 'method')
#8 /projectpath/vendor/twig/twig/src/Template.php(405): __TwigTemplate_a6c3bb7229b7484623fc81785a898df974d3b606abae600aad40adbca18513c0->doDisplay(Array, Array)
#9 /projectpath/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#10 /projectpath/vendor/twig/twig/src/Template.php(390): Twig\Template->display(Array)
#11 /projectpath/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#12 /projectpath/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#13 /projectpath/vendor/craftcms/cms/src/web/View.php(389): Twig\Environment->render('tab', Array)
#14 /projectpath/vendor/barrelstrength/sprout-forms/src/web/twig/variables/SproutFormsVariable.php(154): craft\web\View->renderTemplate('tab', Array)
#15 /projectpath/vendor/twig/twig/src/Extension/CoreExtension.php(1499): barrelstrength\sproutforms\web\twig\variables\SproutFormsVariable->displayTab(Object(barrelstrength\sproutforms\elements\Form), 1950, Array)
#16 /projectpath/vendor/craftcms/cms/src/helpers/Template.php(106): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(barrelstrength\sproutforms\web\twig\variables\SproutFormsVariable), 'displayTab', Array, 'method', false, false)
#17 /projectpath/storage/runtime/compiled_templates/cc/cc418df84888a5d8512bdf811e580e9338bc2ada7113a9a307554fb81cfa787b.php(205): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(barrelstrength\sproutforms\web\twig\variables\SproutFormsVariable), 'displayTab', Array, 'method')
#18 /projectpath/vendor/twig/twig/src/Template.php(405): __TwigTemplate_03fc2c96971de0d3b8bfdd9ebc5b645b7f898c80e40064fccf739b03f0d187d2->doDisplay(Array, Array)
#19 /projectpath/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#20 /projectpath/vendor/twig/twig/src/Template.php(390): Twig\Template->display(Array)
#21 /projectpath/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#22 /projectpath/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#23 /projectpath/vendor/craftcms/cms/src/web/View.php(389): Twig\Environment->render('form', Array)
#24 /projectpath/vendor/barrelstrength/sprout-forms/src/web/twig/variables/SproutFormsVariable.php(94): craft\web\View->renderTemplate('form', Array)
#25 /projectpath/vendor/twig/twig/src/Extension/CoreExtension.php(1499): barrelstrength\sproutforms\web\twig\variables\SproutFormsVariable->displayForm('frostabrygge')
#26 /projectpath/vendor/craftcms/cms/src/helpers/Template.php(106): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(barrelstrength\sproutforms\web\twig\variables\SproutFormsVariable), 'displayForm', Array, 'method', false, false)
#27 /projectpath/storage/runtime/compiled_templates/8d/8d1b58582824245ddd704d916b0b7098fbfc292622fbb825a23b4d9799415b5f.php(327): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(barrelstrength\sproutforms\web\twig\variables\SproutFormsVariable), 'displayForm', Array, 'method')
#28 /projectpath/vendor/twig/twig/src/Template.php(405): __TwigTemplate_5bb961ee433c7596f1e3e58458ef1f6110224a5dfc04647db8e45da8dfcf6f6b->doDisplay(Array, Array)
#29 /projectpath/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#30 /projectpath/vendor/twig/twig/src/Template.php(390): Twig\Template->display(Array)
#31 /projectpath/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#32 /projectpath/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#33 /projectpath/vendor/craftcms/cms/src/web/View.php(389): Twig\Environment->render('_partials/_skje...', Array)
#34 /projectpath/vendor/craftcms/cms/src/web/View.php(450): craft\web\View->renderTemplate('_partials/_skje...', Array)
#35 /projectpath/vendor/craftcms/cms/src/web/Controller.php(241): craft\web\View->renderPageTemplate('_partials/_skje...', Array, 'site')
#36 /projectpath/vendor/craftcms/cms/src/controllers/TemplatesController.php(100): craft\web\Controller->renderTemplate('_partials/_skje...', Array)
#37 [internal function]: craft\controllers\TemplatesController->actionRender('_partials/_skje...', Array)
#38 /projectpath/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#39 /projectpath/vendor/yiisoft/yii2/base/Controller.php(180): yii\base\InlineAction->runWithParams(Array)
#40 /projectpath/vendor/craftcms/cms/src/web/Controller.php(190): yii\base\Controller->runAction('render', Array)
#41 /projectpath/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('render', Array)
#42 /projectpath/vendor/craftcms/cms/src/web/Application.php(274): yii\base\Module->runAction('templates/rende...', Array)
#43 /projectpath/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#44 /projectpath/vendor/craftcms/cms/src/web/Application.php(259): yii\web\Application->handleRequest(Object(craft\web\Request))
#45 /projectpath/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#46 /projectpath/public/index.php(22): yii\base\Application->run()
#47 {main}

```

**Comment by BenParizek** [comment]
@espensgr Can you confirm that by "Article Field" you mean the default Entries Relation field provided by Sprout Forms and not a custom field on your project? And, if that's the case, can you share a screenshot or description of the settings you have for that field?

**Comment by espensgr** [comment]
Sorry, had written an answer here but forgot to comment it 🤦 

@BenParizek yeah, it is is the Entries Relation field
**Craft 2:**
![Skjermbilde 2021-03-25 kl  10 33 49](https://user-images.githubusercontent.com/3862891/112451550-eb0a7600-8d55-11eb-8ab6-eace4eb2977d.png)

**Craft 3:**
![Skjermbilde 2021-03-25 kl  10 34 39](https://user-images.githubusercontent.com/3862891/112451627-02e1fa00-8d56-11eb-81b0-1997511f9c74.png)

"Unike sider" is the same as "unique pages/singles".

On some other note, when i try to add this field as a hidden input in the form, as the user are not adding this themself, it is not added to the submittion:
`<input name="fields[prosjekt]" type="hidden" value="[{{ entryID }}]">`

**Comment by BenParizek** [comment]
@espensgr I've opened a new ticket with your final comment on the hidden field. Let's keep each ticket to one topic so it's easier to keep track of where things stand. I'll follow up on that ticket when I have more.

--

Regarding this item, I'm unable to recreate this error. Do you happen to be using any custom form templates on your site? Is it possible this custom template was copied at one point earlier in the project and some changes have happened since then? What setting are you using for `Settings->Default Form Templates` or `Form Name->Settings->Templates->Templates`?

To troubleshoot further I may need a better understanding of how the data for this form is stored in your database and how your templates are setup. If you can share anything more there, feel free to send it over to sprout@barrelstrengthdesign.com

**Comment by espensgr** [comment]
Yeah, good call.

--

I get the error on the default displayForm tag in twig `{{craft.sproutForms.displayForm(handle)}}`, so no custom template here. And in Craft 3 i guess it defaults to the Sprout default template?

**Comment by BenParizek** [comment]
@espensgr Depending on when you started using Sprout Forms the Form Template may be different. I'd recommend using the "Accessible Templates" option if you can, as we are deprecated the initial "Basic Templates".

I might be missing something still. How is it that you are not using custom templates, but you are also running into the above issue where the Entries Relations field code in your form is custom?

Would you be able to share your database with sprout@barrelstrengthdesign.com and if not, perhaps we can arrange a short screen share this coming week to get to the bottom of this.

**Comment by espensgr** [comment]
Okey, good to know.
It has been a switch from the manual to the `displayForm` at some point, and thing we used the custom code only to override the Entries Relations field with JS before the submit.
I have solved this by manuell setup of the form as we had to use some other custom stuff to we don't use the `displayForm` tag anymore. So we can close this and open it if someone else encounter this. Thanks anyway
